engine/src/main/java/io/camunda/zeebe/streamprocessor/ProcessingStateMachine.java

```diff
@@ -326,6 +326,7 @@ private void errorHandlingInTransaction(final Throwable processingException) thr
           final long position = typedCommand.getPosition();
           final ProcessingResultBuilder processingResultBuilder =
               new DirectProcessingResultBuilder(context, position);
+          processingResultBuilder.reset();
```

Comment:
:wrench: Please add a TODO to revisit this line. I think Chris wants to remove the reset method altogether. And once we have the new implementation, which is based on buffered writers, the reset will no longer be necessary, as creating a new `ProcessingResultBuilder` will also create a new, empty buffer

Reply:
I've opened #10047 and referred to it in a todo